### PR TITLE
COMMERCE-10639 Improve SEO for productFriendlyURL

### PR DESCRIPTION
**Pull request by brianikim**
https://issues.liferay.com/browse/COMMERCE-10639

@andrea-ale-sbarra

### CODE REVIEW

**Comment by brianikim** [comment]
ci:test:relevant

**Comment by brianikim** [comment]
ci:test:commerce

**Comment by brianikim** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/layout/admin/util/CPDefinitionSitemapURLProvider.java [hunk 1]
```diff
@@ -161,13 +168,35 @@ protected void visitLayout(
 				_portal.getClassNameId(CProduct.class),
 				cpDefinition.getCProductId());
 
+		List<FriendlyURLEntryLocalization> friendlyURLEntryLocalizations =
+			_friendlyURLEntryLocalService.getFriendlyURLEntryLocalizations(
+				friendlyURLEntry.getFriendlyURLEntryId());
+
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/layout/admin/util/CPDefinitionSitemapURLProvider.java:174]: I thnk you should move this declaration close to the for loop

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/layout/admin/util/CPDefinitionSitemapURLProvider.java [hunk 2]
```diff
@@ -161,13 +168,35 @@ protected void visitLayout(
...
-			element, productFriendlyURL, typeSettingsUnicodeProperties,
-			layout.getModifiedDate(), productFriendlyURL, null);
+		Map<Locale, String> alternateFriendlyURLs = new HashMap<>();
+
+		for (FriendlyURLEntryLocalization friendlyURLEntryLocalization :
+				friendlyURLEntryLocalizations) {
+
+			String alternateFriendlyURL =
+				currentSiteURL + urlSeparator +
+					friendlyURLEntryLocalization.getUrlTitle();
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/layout/admin/util/CPDefinitionSitemapURLProvider.java:186]: why you didn't use the StringBundler?

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/layout/admin/util/CPDefinitionSitemapURLProvider.java (hunk 2 above)
Comment (brianikim) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/layout/admin/util/CPDefinitionSitemapURLProvider.java:186]: Hi @andrea-ale-sbarra , I was just following existing format in the code. I can update to StringBundler.

Comment (brianikim) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/layout/admin/util/CPDefinitionSitemapURLProvider.java:186]: Made the changes and updated PR.

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/layout/admin/util/CPDefinitionSitemapURLProvider.java (hunk 1 above)
Comment (brianikim) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/layout/admin/util/CPDefinitionSitemapURLProvider.java:179]: Made the requested change and updated PR.

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/layout/admin/util/CPDefinitionSitemapURLProvider.java [hunk 3]
```diff
@@ -161,13 +169,35 @@ protected void visitLayout(
...
+		String productFriendlyURL = StringBundler.concat(
+			currentSiteURL, urlSeparator,
+			friendlyURLEntry.getUrlTitle(themeDisplay.getLanguageId()));
+
+		List<FriendlyURLEntryLocalization> friendlyURLEntryLocalizations =
+			_friendlyURLEntryLocalService.getFriendlyURLEntryLocalizations(
+				friendlyURLEntry.getFriendlyURLEntryId());
+
+		Map<Locale, String> alternateFriendlyURLs = new HashMap<>();
+
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/layout/admin/util/CPDefinitionSitemapURLProvider.java:181]:
```suggestion
		Map<Locale, String> alternateFriendlyURLs = new HashMap<>();
		
		List<FriendlyURLEntryLocalization> friendlyURLEntryLocalizations =
			_friendlyURLEntryLocalService.getFriendlyURLEntryLocalizations(
				friendlyURLEntry.getFriendlyURLEntryId());

```

I meant in this order, https://liferay.atlassian.net/wiki/spaces/ENGR/pages/1511752519/Java+conventions+in+Liferay#%5BinlineExtension%5DLocal-variable-declarations

Comment (brianikim) [reply modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/layout/admin/util/CPDefinitionSitemapURLProvider.java]: @andrea-ale-sbarra , thank you for letting me know about that page. I will take a deep look at it. I updated the PR accordingly.

**Review by andrea-ale-sbarra — APPROVED** [review]
thank you @brianikim

**Comment by andrea-ale-sbarra** [comment]
ci:test:relevant

**Comment by luca-pellizzon** [comment]
ci:test:relevant

**Comment by victoroscuro** [comment]
ci:test:commerce-functional

**Comment by Brunno0liveira** [comment]
Hi @brianikim can you please close this PR

I open a new one https://github.com/liferay-commerce/liferay-portal/pull/3399 for this ticket

Thanks!

**Pull request closed without merging**